lib/PoolClusterFarm.js

```diff
@@ -0,0 +1,120 @@
+var Pool          = require('./Pool');
+var PoolConfig    = require('./PoolConfig');
+var PoolNamespace = require('./PoolNamespace');
+var PoolSelector  = require('./PoolSelector');
+var Util          = require('util');
+var EventEmitter  = require('events').EventEmitter;
+
+/**
+ * PoolClusterFarm
+ * @constructor
+ * @param {object} [config] The pool cluster farm configuration
+ * @public
+ */
+function PoolClusterFarm(config) {
+  EventEmitter.call(this);
+
+  config = config || {};
+  this._servers = [];
+  this._numMasters = 0;
+  this._numSlaves = 0;
+  this._last_index = 0;
+  this.debug = config.debug || false
+}
+
+Util.inherits(PoolClusterFarm,EventEmitter);
+
+PoolClusterFarm.prototype.query = function query(sql,values,cb) {
+	var self = this
+
+	if (!this._numMasters || this._numMasters < 1) {
+		if (cb && typeof cb == 'function') { return cb('No servers assigned to PoolClusterFarm. Please verify...'); }
+		else { return values('No servers assigned to PoolClusterFarm. Please verify...'); }
+	}
+
+	if (typeof sql != 'string' || sql.length < 7) {
+		if (cb && typeof cb == 'function') { return cb('Invalid sql statement. Please verify...'); }
+		else { return values('Invalid sql statement. Please verify...'); }
+	}
+
+	if (sql.substr(0,6).toLowerCase() == 'select') {
```

Comment:
Well, you would have one object, pool.query(), and it would allow for you to have unlimited scaling. For example, if you had a large app that had millions of users, one database server probably can not handle that much traffic. That being said, a cluster of master servers can handle all those requests pretty easily and it would scale. Sorry if I'm not being clear.